- made dependency parameter aware of multiple types

LymanZerga opened the pull request:
For some of our NetArchTests we would like to violate dependecies of multiple types in one rule. As the internal search engine already provide a list of dependencies, this change just enables this on client side. With the params keyword we do no need to change the signature in terms of creating 2 overloads.

BenMorris commented:
Hi,

I have had a look at a fuller implementation that implements this as a predicate, avoids breaking change and adds unit tests.

Its on this branch: https://github.com/BenMorris/NetArchTest/tree/HaveDependencyOnMultiple

The one catch is that the types are matched if they have a dependency on _any_ item in the list. I am trying to decide whether this is a problem. I guess if you want to check for types that meet a list of dependencies you would use successive `And()` operators. 

I will need to do a bit more testing to make sure that I'm not introducing something ambiguous into the API.

LymanZerga commented:
Hi,

you are right. In our concrete case we want to match if any of the type in the list has a dependecy to the type under test. So better to revert this change and offer two more methods HaveDependecyOnAny and HaveDependencyOnAll which maybe internaly uses the` And()/Or()` operators and are much more epressiv.

LymanZerga commented:
Hi,

I have added methods for multiple dependecies with any and all. I have also added several unit tests for that, all have passed. This should not be ambiguous anymore since the methods are very expressive.
